lib/diff.ml

```diff
@@ -107,34 +96,29 @@ let value_item ~typing_env ~name ~reference ~current =
       | exception Includecore.Dont_match _ ->
           Some (Value { vname = name; vdiff = Modified { reference; current } })
       )
-  | _ -> None
 
 let rec items ~reference ~current =
   let env = Typing_env.for_diff ~reference ~current in
   let ref_items = extract_items reference in
   let curr_items = extract_items current in
-  Sig_item_map.merge
-    (fun (item_type, name) ref_opt curr_opt ->
-      match (item_type, ref_opt, curr_opt) with
-      | Value_item, reference, current ->
-          value_item ~typing_env:env ~name ~reference ~current
-      | Module_item, reference, current ->
-          module_item ~typing_env:env ~name ~reference ~current
-      | Type_item, reference, current ->
-          type_item ~typing_env:env ~name ~reference ~current)
-    ref_items curr_items
-  |> Sig_item_map.bindings |> List.map snd
+  let diff_item : type a. (a, 'diff) Sig_item_map.diff_item =
+   fun item_type name reference current ->
+    match item_type with
+    | Sig_item_map.Value -> value_item ~typing_env:env ~name ~reference ~current
```

Comment:
Are those `Sig_item_map.` prefix path needed? The compiler should not need them, it knows the type of `item_type`. At least, the second and third one can be removed.